system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,98 +28,186 @@ The application package has the following folder structure:
-
-> **Note**
-> The format of the application description is *inspired by* the [Open Application Model](https://github.com/oam-dev/spec) (OAM), however, the format defined here is a *narrow subset*, which is sufficient for the goals of the Margo specification at this point.
+The `margo.yaml` file is the application description. The purpose of this file is to present the application on an application catalog or marketplace from where an end user selects the application to hand it over to the Workload Orchestration Software, which configures it and makes it available for installation on the edge device (see Section [Workload Orchestration Agent](./workload-orchestration-agent.md)).
 
 **Application Description Example**
 
-An example of an `margo.yaml` file is shown below:
+A simple hello-world example of an `margo.yaml` file is shown below:
 
 ```yaml
-apiVersion: margo.dev/v1 
-kind: application 
-metadata: 
-  name: my-helm-nginx-margo-app 
-  version: 0.1.0 
-  description: Example of packaging nginx as a Margo application.
-  icon: "./resources/nginx-logo-s.png" 
-  author: John Smith 
-  organization: App Devs Inc. 
-  organization-site: https://app-devs.com 
-  app-tagline: My cool example app. 
-  description-long: Long description of my cool app ... 
-  license-file: "./resources/LICENCE.txt" 
-  app-site: https://my-cool-app.com 
-spec: 
-  helm-chart: 
-    repoType: local
-    url: "./manifest/helm-chart" 
-  docker-compose: 
-    repoType: local 
-    url: "./manifest/docker-compose/compose.yaml" 
+apiVersion: margo.org/v1-alpha1
+kind: application
+metadata:
+  id: com.northstartida.hello.world
+  name: Hello World
+  description: A basic hello world application
+  version: 1.0
+  catalog:
+    application:
+      icon: ./resources/hw-logo.png
+      tagline: Northstar Industrial Application's hello world application.
+      descriptionFile: ./resources/description.md
+      releaseNotes: ./resources/release-notes.md
+      licenseFile: ./resources/license.pdf
+      site: http://www.northstar-ida.com
+    author:
+      - name: Roger Wilkershank
+        email: rpwilkershank@northstar-ida.com
+    organization:
+      - name: Northstar Industrial Applications
+        site: http://northstar-ida.com
+components:
+  cluster:
+    - name: hello-world
+      type: helm.v3
+      properties:  
+        repository: oci://northstarida.azurecr.io/charts/hello-world
+        revision: 1.0.1
+        wait: true
+```
+
+An example of a `margo.yaml` file using multiple helm charts and a docker-compose file package is shown below.
+
+```yaml
+apiVersion: margo.org/v1-alpha1
+kind: application
+metadata:
+  id: com.northstartida.digitron.orchestrator
+  name: Digitron orchestrator
+  description: The Digitron orchestrator application
+  version: 1.2.1 
+  catalog:
+    application:
+      icon: ./resources/ndo-logo.png
+      tagline: Northstar Industrial Application's next-gen, AI driven, Digitron instrument orchestrator.
```

Comment:
If tagline has a list nature, it should be also specified as a list of tags. Else, people might put sentences here.

Reply:
tagline is supposed to be a sentence. It's a [marketing tagline/slogan](https://www.feedough.com/tagline-meaning-examples-ideas-how-to-guide/) not a list of tags

Reply:
Yes, makes senses. Especially as you need something, which you put next to the icon in the catalog. However,  a list of tags would also be helpful, e.g. to map an app to defined categories like optimization, instrumentation, monitoring etc.

Reply:
We agreed in the telco to include a list of "tags".

Reply:
tags was added to the specification.